elastic/synthetics-recorder
# chore: Remove unused exports

## Summary

Resolves #188.

We have a lot of unused exports in the recorder, our current linting setup does not catch these.

This enabled me to delete a bunch of dead code.

## Implementation details

This adds a new dev dependency, `ts-unused-exports`, and a new script that calls it. It can be run using `npm run unused-exports`. The script throws an error when unused exports are detected.

This will have a slightly annoying side effect of calling out exports that aren't referenced outside of the immediate file, but _do_ have references in the same file. In this case these symbols shouldn't _technically_ be exported, but are still used internally. This rule might be a little draconian, but I think it's worthwhile to prevent the existence of entire files and other objects that are completely dead.

I'm open to improvement suggestions, if people think this is too strong of an imposition.

## How to validate this change

You can create an unused export in any `ts`/`tsx` file, like `export type Unused = "nothing"` and run the script, it should throw with a reference to the filename and offending line.

I've also updated the CI to run this script, I will be debugging that before requesting review from others.
